### ISSUE TITLE
Illegal instruction in ClusterGCNSampler

### ISSUE THREAD

**Issue by gpzlx1** [dmlc/dgl#3910]
## To Reproduce

Just run the following python code

```python
import dgl
import torch
from dgl.data import RedditDataset

def load_reddit():
    data = RedditDataset(self_loop=True)
    g = data[0]
    n_classes = data.num_classes
    train_nid = torch.nonzero(g.ndata["train_mask"], as_tuple=True)[0]
    test_nid = torch.nonzero(g.ndata["test_mask"], as_tuple=True)[0]
    val_nid = torch.nonzero(g.ndata["val_mask"], as_tuple=True)[0]
    splitted_idx = {"train":train_nid, "test":test_nid, "val":val_nid}
    feat = g.ndata['feat']
    labels = g.ndata['label']
    g.ndata.clear()
    return g, feat, labels, n_classes, splitted_idx

g, _, _, _, _ = load_reddit()
train_nid = torch.arange(g.num_nodes())
print(g)

sampler = dgl.dataloading.ClusterGCNSampler(g, 1000, cache_path='cluster_gcn.cache')
```

It will raise an `Illegal instruction` error.

```shell
Graph(num_nodes=232965, num_edges=114848857,
      ndata_schemes={}
      edata_schemes={'__orig__': Scheme(shape=(), dtype=torch.int64)})
Illegal instruction
```

In fact, `metis_partition_assignment` in `ClusterGCNSampler` raises the error.
It seems that the repo in the `data.dgl.ai` is compiled with the unusual instruction set that the consumer-grade CPU does not support.

## Expected behavior

No errors

## Environment

 - DGL Version : `0.8.0post2`, installed by  `pip install dgl-cu113 dglgo -f https://data.dgl.ai/wheels/repo.html --no-deps`
 - Backend Library & Version : `1.11.0+cu113`, installed by `pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu113`

  

## Additional context

I've tested it on two platforms, both go wrong:
Amd 3600 + Geforce 3080 12G, cuda11.6
```shell
$ lscpu
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl tsc_reliable nonstop_tsc cpuid extd_apicid pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext ssbd ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves clzero xsaveerptr arat npt nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold v_vmsave_vmload umip rdpid
```

E5-2620 v4 @ 2.10GHz + Geforce 1080Ti 11G, cuda11.1
 ```shell
$ lscpu
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap intel_pt xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts md_clear flush_l1d
```

**Comment by BarclayII** [comment]
Turns out that CMakeLists in METIS uses `-march=native`.  Since we migrated the build server to a g4 instance, the compiled code may not be able to run on older CPUs.

**Comment by gpzlx1** [comment]
@BarclayII Sorry for not timely reply. The problem still exists
```python3
$ virtualenv --python=pyhon3 test_venv
$ source test_venv/bin/activate
$ pip install torch
$ pip install --pre dgl -f https://data.dgl.ai/wheels-test/repo.html -no-deps
$ pip install networkx psutil requests scipy tqdm
$ python3 test.py
Illegal instruction
```

**Comment by gpzlx1** [comment]
@BarclayII @jermainewang The issue has been fixed on the latest DGL (0.9). Thx a lot.
